x/data/client/testsuite/grpc.go

```diff
@@ -159,6 +161,249 @@ func (s *IntegrationTestSuite) TestQueryHashByIRI() {
+				val.APIAddress,
+				ch.Graph.DigestAlgorithm,           // enum 1
+				ch.Graph.CanonicalizationAlgorithm, // enum 1
+			),
+			true,
+			"hash cannot be empty",
+		},
+		{
+			"invalid hash",
+			fmt.Sprintf(
+				"%s/regen/data/v1/iri-by-graph?hash=%s&digest_algorithm=%d&canonicalization_algorithm=%d",
+				val.APIAddress,
+				"foo",
+				ch.Graph.DigestAlgorithm,           // enum 1
+				ch.Graph.CanonicalizationAlgorithm, // enum 1
+			),
+			true,
+			"failed to decode base64 string",
+		},
+		{
+			"unspecified digest algorithm",
+			fmt.Sprintf(
+				"%s/regen/data/v1/iri-by-graph?hash=%s&canonicalization_algorithm=%d",
+				val.APIAddress,
+				encodedHash,                        // base64 encoded string
+				ch.Graph.CanonicalizationAlgorithm, // enum 1
+			),
+			true,
+			"digest algorithm cannot be unspecified",
+		},
+		{
+			"invalid digest algorithm",
+			fmt.Sprintf(
+				"%s/regen/data/v1/iri-by-graph?hash=%s&digest_algorithm=%s&canonicalization_algorithm=%d",
+				val.APIAddress,
+				encodedHash, // base64 encoded string
+				"foo",
+				ch.Graph.CanonicalizationAlgorithm, // enum 1
+			),
+			true,
+			"foo is not a valid data.DigestAlgorithm",
+		},
+		{
+			"unspecified canonicalization algorithm",
+			fmt.Sprintf(
+				"%s/regen/data/v1/iri-by-graph?hash=%s&digest_algorithm=%s",
+				val.APIAddress,
+				encodedHash,              // base64 encoded string
+				ch.Graph.DigestAlgorithm, // enum 1
+			),
+			true,
+			"canonicalization algorithm cannot be unspecified",
+		},
+		{
+			"invalid canonicalization algorithm",
+			fmt.Sprintf(
+				"%s/regen/data/v1/iri-by-graph?hash=%s&digest_algorithm=%s&canonicalization_algorithm=%s",
+				val.APIAddress,
+				encodedHash,              // base64 encoded string
+				ch.Graph.DigestAlgorithm, // enum 1
+				"foo",
+			),
+			true,
+			"foo is not a valid data.GraphCanonicalizationAlgorithm",
+		},
+		{
+			"valid request",
+			fmt.Sprintf(
+				"%s/regen/data/v1/iri-by-graph?hash=%s&digest_algorithm=%d&canonicalization_algorithm=%d",
+				val.APIAddress,
+				encodedHash,                        // base64 encoded string
+				ch.Graph.DigestAlgorithm,           // enum 1
+				ch.Graph.CanonicalizationAlgorithm, // enum 1
+			),
+			false,
+			"",
+		},
+		{
+			"valid request enums as strings",
+			fmt.Sprintf(
+				"%s/regen/data/v1/iri-by-graph?hash=%s&digest_algorithm=%s&canonicalization_algorithm=%s",
+				val.APIAddress,
+				encodedHash,                    // base64 encoded string
+				"DIGEST_ALGORITHM_BLAKE2B_256", // enum 1
+				"GRAPH_CANONICALIZATION_ALGORITHM_URDNA2015", // enum 1
+			),
+			false,
+			"",
+		},
+	}
+
+	require := s.Require()
+	for _, tc := range testCases {
+		tc := tc
+		s.Run(tc.name, func() {
+			bz, err := rest.GetRequest(tc.url)
+			require.NoError(err)
+
+			var res data.QueryIRIByGraphHashResponse
+			err = val.ClientCtx.Codec.UnmarshalJSON(bz, &res)
+
+			if tc.expErr {
+				require.Error(err)
+				require.Contains(string(bz), tc.errMsg)
```

Comment:
ditto

Reply:
Same as above. We need to check if the response contains the error message.